runtime/lua/vim/treesitter/dev.lua

```diff
@@ -501,12 +501,21 @@ function M.inspect_tree(opts)
     end,
   })
 
-  api.nvim_create_autocmd('BufHidden', {
+  api.nvim_create_autocmd({ 'WinClosed' }, {
     group = group,
     buffer = buf,
-    once = true,
     callback = function()
-      close_win(w)
+      local windows = vim.fn.win_findbuf(buf) --[=[@as integer[]]=]
+      if #windows == 1 then
+        close_win(w)
+        return true
+      end
+      for _, new_win in pairs(windows) do
+        if new_win ~= win then
+          win = new_win
+          break
+        end
+      end
```

Comment:
finding an item is a good use-case for vim.iter

Reply:
I would do `vim.iter(pairs(windows))`. ArrayIter doesn't really help with find AFAIK

Reply:
> I would do `vim.iter(pairs(windows))`. ArrayIter doesn't really help with find AFAIK

Done. Despite not really knowing the difference 😅. Would appreciate it if you could explain it or point me somewhere though because I could only find mentions of `vim.iter(ipairs(...))` in the docs.